local.tf

```
@@ -32,6 +32,7 @@ locals {
     asg_force_delete              = false                       # Enable forced deletion for the autoscaling group.
     asg_initial_lifecycle_hooks   = []                          # Initital lifecycle hook for the autoscaling group.
     asg_recreate_on_change        = false                       # Recreate the autoscaling group when the Launch Template or Launch Configuration change.
+    asg_name_prefix               = null                        # First part of autoscaling groups' names. By default, it will be set to cluster name.
```

```suggestion
    asg_name_prefix               = local.default_asg_name_prefix                        # First part of autoscaling groups' names. By default, it will be set to cluster name.
```

And define a local variable like this :
```hcl
default_asg_name_prefix = concat(aws_eks_cluster.this.*.name, [""])[0]
```

In that way, the module won't fail if `create_eks` is false.